diff -up gpgme-1.8.0/lang/cpp/src/GpgmeppConfig.cmake.in.in.extras gpgme-1.8.0/lang/cpp/src/GpgmeppConfig.cmake.in.in
--- gpgme-1.8.0/lang/cpp/src/GpgmeppConfig.cmake.in.in.extras	2017-01-18 11:28:03.707578717 -0600
+++ gpgme-1.8.0/lang/cpp/src/GpgmeppConfig.cmake.in.in	2017-01-18 11:33:05.274758962 -0600
@@ -63,7 +63,7 @@ add_library(Gpgmepp SHARED IMPORTED)
 
 set_target_properties(Gpgmepp PROPERTIES
   INTERFACE_INCLUDE_DIRECTORIES "@resolved_includedir@/gpgme++;@resolved_includedir@"
-  INTERFACE_LINK_LIBRARIES "pthread;@resolved_libdir@/libgpgme.so;@LIBASSUAN_LIBS@"
+  INTERFACE_LINK_LIBRARIES "pthread;@resolved_libdir@/libgpgme.so"
   IMPORTED_LOCATION "@resolved_libdir@/libgpgmepp.so"
 )
 
